File: app/src/main/java/com/bmn/bookfinder/data/network/remote/ApiClient.java
package com.bmn.bookfinder.data.network.remote;

import android.content.Context;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiClient {

    public static Retrofit retrofit = null;

    public static Retrofit getApi(Context context, String baseUrl) {

        Gson gson = new GsonBuilder()
                .setLenient()
                .create();

        OkHttpClient okHttpClient;

        okHttpClient = new OkHttpClient.Builder().addInterceptor(chain -> {
            Request.Builder builder = chain.request().newBuilder();
            builder.addHeader("Content-Type", "application/json");

            builder.method(chain.request().method(), chain.request().body());
            return chain.proceed(builder.build());
        }).readTimeout(120, TimeUnit.SECONDS).connectTimeout(120, TimeUnit.SECONDS).build();

        retrofit = new Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .client(okHttpClient)
                .build();

        return retrofit;
    }

}

File: app/src/main/java/com/bmn/bookfinder/data/network/remote/ApiClient.kt
package com.bmn.bookfinder.data.network.remote

import android.content.Context
import com.google.gson.GsonBuilder
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

object ApiClient {
    private  lateinit var retrofit: Retrofit
    fun getApi(context: Context?, baseUrl: String): Retrofit {
        val gson = GsonBuilder()
            .setLenient()
            .create()
        val okHttpClient: OkHttpClient =
            OkHttpClient.Builder()
                .addInterceptor { chain: Interceptor.Chain ->
                    val builder = chain.request().newBuilder()
                    builder.addHeader("Content-Type", "application/json")
                    builder.method(chain.request().method(), chain.request().body())
                    chain.proceed(builder.build())
                }
                .readTimeout(120, TimeUnit.SECONDS)
                .connectTimeout(120, TimeUnit.SECONDS)
                .build()
        retrofit = Retrofit.Builder()
            .baseUrl(baseUrl)
            .addConverterFactory(GsonConverterFactory.create(gson))
            .client(okHttpClient)
            .build()
        return retrofit
    }
}